lib/plugins/auto-load.js

```diff
@@ -3,40 +3,44 @@
 const _ = require('lodash');
 const debug = require('debug')('voxa');
 
+let defaultConfig = { loadByToken: false };
+
 function autoLoad(skill, config) {
-  skill.onRequestStarted((alexaEvent) => {
-    if (!config) {
-      throw Error('Empty config file');
-    }
+  if (!config) {
+    throw Error('Empty config file');
+  }
 
-    if (!config.adapter) {
-      throw Error('Empty adapter');
-    }
+  if (!config.adapter) {
+    throw Error('Empty adapter');
+  }
+
+  if (!_.isFunction(config.adapter.get)) {
+    throw Error('No get method to fetch data from');
+  }
 
-    if (config.loadByToken && !_.get(alexaEvent, 'session.user.accessToken')) {
+  defaultConfig = _.merge(defaultConfig, config);
+
+  skill.onRequestStarted((alexaEvent) => {
+    if (defaultConfig.loadByToken && !_.get(alexaEvent, 'session.user.accessToken')) {
       debug('Missing token on load data by token operation');
       return alexaEvent;
     }
 
     let key = _.get(alexaEvent, 'user.userId');
 
-    if (config.loadByToken) {
+    if (defaultConfig.loadByToken) {
```

Comment:
this should probably just be a part of the adapter, not the plugin. just pass alexaEvent.user to the adapter and have the adapter figure out what to do